### PR TITLE
bwbasic: darwin build fix, derivation cleanup / RFC format

### PR DESCRIPTION
**Pull request by matteo-pacini**
## Description of changes

- Forcing GCC compiler on Darwin too
- Modernised the derivation (`with lib;` / `hash` and RFC format)

Screenshot 2024-08-23 at 09 55 26

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 336730` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bwbasic</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
-stdenv.mkDerivation rec {
+{
+  lib,
+  gccStdenv,
+  dos2unix,
+  fetchurl,
+  unzip,
+}:
+
+gccStdenv.mkDerivation {
```
Comment (smancill) [inline pkgs/development/interpreters/bwbasic/default.nix:9]:
```suggestion
gccStdenv.mkDerivation (finalAttrs: {
```

See below comment.

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
   pname = "bwbasic";
   version = "3.20";
 
   src = fetchurl {
     url = "mirror://sourceforge/project/bwbasic/bwbasic/version%203.20/bwbasic-3.20.zip";
```
Comment (smancill) [inline pkgs/development/interpreters/bwbasic/default.nix:14]:
```suggestion
    url = "mirror://sourceforge/project/bwbasic/bwbasic/version%20${finalAttrs.version}/bwbasic-${finalAttrs.version}.zip";
```

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 3]
```diff
@@ -27,12 +36,12 @@ stdenv.mkDerivation rec {
...
     mainProgram = "bwbasic";
-    license = licenses.gpl2Only;
-    maintainers = with maintainers; [ irenes ];
-    platforms = platforms.all;
+    license = lib.licenses.gpl2Only;
+    maintainers = with lib.maintainers; [ irenes ];
+    platforms = lib.platforms.all;
     homepage = "https://sourceforge.net/projects/bwbasic/";
   };
 }
```
Comment (smancill) [inline pkgs/development/interpreters/bwbasic/default.nix:47]:
```suggestion
})
```

**Review by smancill — COMMENTED** [review]
I think that it may be better to use `version` in the url when doing the derivation cleanup, thus also changing to `finalAttrs`.

But someone may think that's not really necessary, so I am not 100% sure.

**Comment by matteo-pacini** [comment]
@smancill Thanks for the feedback, changes done but your comments do not refresh for some reason (?):

Here's the latest version:
https://github.com/NixOS/nixpkgs/pull/336730/files

**Pull request merged**